### PR TITLE
feat: adding event clickawaylistener

### PR DESCRIPTION
**Pull request by panvourtsis**
## Description

This PR adds the event on clickAwayListner click

### CODE REVIEW

File: src/components/utils/ClickAwayListener/ClickAwayListener.tsx [hunk 1]
```diff
@@ -1,18 +1,23 @@
 import * as React from 'react';
+import { useEffect } from 'react';
 import { ReactFCC } from 'utils/types';
 
-const useClickAwayListener = (ref: React.MutableRefObject<any>, onClick: () => void) => {
-  const handleClickOutside = (event: any) => {
-    if (ref.current && !ref.current.contains(event.target)) {
-      onClick();
+const useClickAwayListener = (
+  ref: React.MutableRefObject<HTMLElement | null>,
```
Comment (mkarajohn) [inline src/components/utils/ClickAwayListener/ClickAwayListener.tsx:6]: Import `type MutableRefObject` directly from `'react'`

**Review by mkarajohn — CHANGES_REQUESTED** [review]

File: src/components/utils/ClickAwayListener/ClickAwayListener.tsx [hunk 2]
```diff
@@ -1,18 +1,23 @@
...
+  onClick: (event: MouseEvent) => void,
+  useCapture = false
```
[the remark this answers is not in the document]
Comment (panvourtsis) [reply src/components/utils/ClickAwayListener/ClickAwayListener.tsx:8]: i do use the official one, there are two and we use a specific `eslint` rule that makes easier to use the second official one
image

File: src/components/utils/ClickAwayListener/ClickAwayListener.tsx (hunk 1 above)
Comment (panvourtsis) [reply src/components/utils/ClickAwayListener/ClickAwayListener.tsx]: fixed on https://github.com/Orfium/orfium-ictinus/pull/713/commits/5856215cdecd34ea090612a6d94a5467716bff0f

File: src/components/utils/ClickAwayListener/ClickAwayListener.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (panvourtsis) [reply src/components/utils/ClickAwayListener/ClickAwayListener.tsx:8]: default value changed on https://github.com/Orfium/orfium-ictinus/pull/713/commits/5856215cdecd34ea090612a6d94a5467716bff0f

**Pull request merged**